documentation/API-Testing-Guidelines.md

```diff
@@ -0,0 +1,41 @@
+# API Testing Guidelines
+
+## Table of Contents
+1. [Introduction](#introduction)
+2. [Test Cases contributions for APIs](#contribution)
+3. [Location of feature file](#location)
+4. [Best practices and recommendations](#recommendations)
+7. [References](#references)
+
+## Introduction <a name="introduction"></a>
+This document captures guidelines for the API testing in CAMARA project. These guidelines are applicable to every API to be worked out under the CAMARA initiative.
+
+## Test Cases contribution for APIs <a name="contribution"></a>
+Based on the decision taken in the commonalities working group and as documented in [API-Readiness-Checklist](https://github.com/camaraproject/Commonalities/blob/main/documentation/API-Readiness-Checklist.md), a Gherkin feature file will be added to the main subproject repo and this will fullfil the minimum criteria of readiness w.r.t API test cases and documentation. If subprojects also intend to add test implementations, an aligned single implementation that is agreed amongst all provider implementers could be added to the main subproject repo. If no alignment is possible, each provider implementer will add the test implementation to their own repos.
```

Comment:
```suggestion
Based on the decision taken in the Commonalities working group and as documented in [API-Readiness-Checklist](https://github.com/camaraproject/Commonalities/blob/main/documentation/API-Readiness-Checklist.md), at least one Gherkin feature file will be added to the main subproject repo and this will fulfill the minimum criteria of readiness with respect to API test cases and documentation. If subprojects also intend to add test implementations, an aligned single implementation that is agreed among all provider implementors could be added to the main subproject repo. If no alignment is possible, each provider implementer will add the test implementation to their own repos.
```

Other comments:

- In some cases we may need several Gherkin files, as in the issue we discussed the convenience to have a .feature file per API operation, specially for APIs with many endpoints.
- Not sure about the convenience to add implementations directly to the subprojects